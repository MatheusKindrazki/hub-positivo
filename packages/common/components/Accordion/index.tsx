import {
  Collapse as Accordion,
  CollapseProps as AccordionProps,
  UnmountClosed
} from 'react-collapse'

export { Accordion, UnmountClosed }
export type { AccordionProps }
