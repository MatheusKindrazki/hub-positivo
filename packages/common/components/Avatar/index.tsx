import React from 'react'

import { Avatar as AvatarChakra, AvatarProps } from '@chakra-ui/react'

const Avatar: React.FC<AvatarProps> = ({ ...rest }) => {
  return (
    <AvatarChakra
      width="2.5rem"
      height="2.5rem"
      backgroundColor="gray.400"
      name="HUB"
      {...rest}
    />
  )
}

export default Avatar
export type { AvatarProps }
