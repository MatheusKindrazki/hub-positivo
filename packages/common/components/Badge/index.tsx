import { Badge, BadgeProps } from '@chakra-ui/react'

export default Badge
export type { BadgeProps }
