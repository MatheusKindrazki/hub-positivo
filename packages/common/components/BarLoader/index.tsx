import React from 'react'

import { BarLoader as Loading } from 'react-spinners'

import { useTheme } from '../../layout'

type BarProps = typeof Loading.defaultProps

const BarLoader: React.FC<BarProps> = ({ loading, ...rest }) => {
  const { colors } = useTheme()

  return (
    <Loading
      color={colors.blue[500]}
      width="100%"
      height="5px"
      loading={loading}
      css={`
        border-radius: 8px;
        position: fixed;
        top: 0
        bottom:0;
      `}
      {...rest}
    />
  )
}

export default BarLoader
