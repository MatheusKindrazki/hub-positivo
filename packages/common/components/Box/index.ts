import { Box, BoxProps } from '@chakra-ui/react'

export type { BoxProps }

export default Box
