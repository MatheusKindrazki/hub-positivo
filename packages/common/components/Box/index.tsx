import React from 'react'

import { Box as BoxChakra, BoxProps as Props } from '@chakra-ui/react'

export interface BoxProps extends Props {
  size?: Props['width']
}

const Box: React.FC<BoxProps> = ({ children, ...rest }) => {
  return <BoxChakra {...rest}>{children}</BoxChakra>
}

export default Box
