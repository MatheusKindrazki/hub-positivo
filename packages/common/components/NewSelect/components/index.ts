export { default as Icon } from './Icon'
export { default as InputSelect } from './InputSelect'
export { default as Control } from './Control'
export { default as NotFound } from './NotFound'
