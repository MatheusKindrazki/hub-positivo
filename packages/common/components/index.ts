export { default as Box } from './Box'
export { default as Text } from './Text'
export { default as Flex } from './Flex'
export { default as Menu } from './Menu'
export { default as Tour } from './Tour'
export { default as Input } from './Input'
export { default as Image } from './Image'
export { default as Stack } from './Stack'
export { default as Badge } from './Badge'
export { default as Modal } from './Modal'
export { default as Select } from './Select'
export { default as Checkbox } from './Checkbox'
export { default as ButtonGroup } from './ButtonGroup'
export { default as Divider } from './Divider'
export { default as Popover } from './Popover'
export { default as Table } from './Table'
export { default as Switch } from './Switch'
export { default as Label } from './Label'
export { default as Avatar } from './Avatar'
export { default as Drawer } from './Drawer'
export { default as Button } from './Button'
export { default as Search } from './Search'
export { default as Tooltip } from './Tooltip'
export { default as CardBox } from './CardBox'
export { default as Welcome } from './Welcome'
export { default as Heading } from './Heading'
export { default as Collapse } from './Collapse'
export { default as Skeleton } from './Skeleton'
export { default as SimpleGrid } from './SimpleGrid'
export { default as BarLoader } from './BarLoader'
export { default as FormInput } from './Form/Input'
export { default as FormSelect } from './Form/Select'
export { default as FormButton } from './Form/Button'
export { default as CardProduct } from './CardProduct'
export { default as FormControl } from './FormControl'
export { default as ProfileItem } from './ProfileItem'
export { default as Breadcrumbs } from './Breadcrumbs'
export { default as List } from './List/List'
export { default as ListItem } from './List/ListItem'
export { default as Link } from './Link'
export { default as SpinnerLoader } from './SpinnerLoader'
