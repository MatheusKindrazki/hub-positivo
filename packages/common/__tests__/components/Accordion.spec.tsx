import React from 'react'

import { render } from '@psdhub/test-utils'

import { Accordion, UnmountClosed } from '../../components/Accordion'

describe('Accordion an UnmountClosed components', () => {
  it('Accordion being rendered on canvas', () => {
    const wrapper = render(<Accordion isOpened>hub</Accordion>)
    expect(wrapper).toMatchSnapshot()
  })

  it('UnmountClosed being rendered on canvas', () => {
    const wrapper = render(<UnmountClosed isOpened>hub</UnmountClosed>)

    expect(wrapper).toMatchSnapshot()
  })
})
