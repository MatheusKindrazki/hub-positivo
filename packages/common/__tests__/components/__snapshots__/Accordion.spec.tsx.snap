// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion an UnmountClosed components Accordion being rendered on canvas 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        aria-hidden="false"
        class="ReactCollapse--collapse"
        style="height: auto; overflow: initial;"
      >
        <div
          class="ReactCollapse--content"
        >
          hub
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="false"
      class="ReactCollapse--collapse"
      style="height: auto; overflow: initial;"
    >
      <div
        class="ReactCollapse--content"
      >
        hub
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Accordion an UnmountClosed components UnmountClosed being rendered on canvas 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        aria-hidden="false"
        class="ReactCollapse--collapse"
        style="height: auto; overflow: initial;"
      >
        <div
          class="ReactCollapse--content"
        >
          hub
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="false"
      class="ReactCollapse--collapse"
      style="height: auto; overflow: initial;"
    >
      <div
        class="ReactCollapse--content"
      >
        hub
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
